### ISSUE TITLE
How to store Microbiology Susceptibility data into OMOP CDM

### ISSUE THREAD

**Issue by crabcakeworld** [OHDSI/Themis#172]
## CDM or THEMIS convention?
THEMIS 

## Table or Field level?
Table

## Is this a general convention?
Applies to multiple fields

## Summary of issues
Below is taken from a real EHR data (PHI removed):
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

mio_src | mio_body_site | cm_organism | specimen | antibiotic | result | result_description | lab test LOINC code
-- | -- | -- | -- | -- | -- | -- | --
Abscess | Hand, Right | Staphylococcus aureus | Abscess | Moxifloxacin | <=0.25 | Sensitive | 31039-1

</body>

</html>

How to put these information into OMOP CDM?

## Summary of answer

This has not been rectified.

When taking above data into OMOP CDM, it will load into 3 records as follows:
1) A Specimen record that contains specimen and anatomic site.
2) A Measurement record that contains microbiology test and test result and point to the first Specimen record.
3) A Measurement record that contains the organism that is being tested and point to the 2nd Measurement record.

The above source data will become following:

**Record 1**
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

specimen_id | specimen_concept_id | anatomic_site_concept_id | disease_status_concept_id
-- | -- | -- | --
34567890 | 34567890 | 4124271	     (Abscess swab) | 4302584     (Structure of right hand) | NULL

</body>

</html>

**Record 2**
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

measurement_id | measurement_concept_id | value_as_Concept_id | value_as_number | unit_concept_id | measurement_source_value | measurement_event_id | meas_event_field_concept_id
-- | -- | -- | -- | -- | -- | -- | --
12345678 | 3037708     (Moxifloxacin [Susceptibility]) | 4307105     (Sensitive) | 0.25 | 8859     (microgram per milliliter) | 31039-1     (Moxifloxacin [Susceptibility]) | 34567890 | 1147049  (specimen.specimen_id)

</body>

</html>

**Record 3**
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/QIBB41~1.YAN/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

measurement_id | measurement_concept_id | value_as_Concept_id | value_as_number | unit_concept_id | measurement_source_value | measurement_event_id | meas_event_field_concept_id
-- | -- | -- | -- | -- | -- | -- | --
23456789 | 4189544	     (Microbial culture) | 4149419	     (Staphylococcus aureus) |   |   |   | 12345678 | 1147729	     (measurement.measurement_id)

</body>

</html>

## Related links

https://forums.ohdsi.org/t/mapping-microbiology-susceptibility-into-omop-cdm4-observations/167/57

## Other comments/notes

The above solution has a problem. In the 3rd record, the value_as_concept_id field only allows concept_id of "Meas Value" domain. But the organism concept_id, 4149419  (Staphylococcus aureus) belongs to Observation domain. There might be other solutions out there. But the above solution is the presiding idea.

**Comment by don-torok** [comment]
The relationships Specimen to Susceptibility then Susceptibility to Organism seems wrong. I think the relationship should be Specimen to Organism then Organism to Susceptibility. So that the Organism record will point back to the specimen and the susceptibility record will point to the organism.

**Comment by crabcakeworld** [comment]
> The relationships Specimen to Susceptibility then Susceptibility to Organism seems wrong. I think the relationship should be Specimen to Organism then Organism to Susceptibility. So that the Organism record will point back to the specimen and the susceptibility record will point to the organism.

Yes, @don-torok. You are correct!. I have modified my design according your comment.

**Comment by crabcakeworld** [comment]
Here is the microbiology data model consolidation file. We can discuss it during our next Themis meeting. 
[Microbiology data models consolidation.xlsx](https://github.com/user-attachments/files/17848812/Microbiology.data.models.consolidation.xlsx)

**Comment by MelaniePhilofsky** [comment]
Connecting @Aew0330

**Comment by MelaniePhilofsky** [comment]
Connecting @AbeckCO   who is also working on this issue

**Comment by AbeckCO** [comment]
@MelaniePhilofsky - Thank you for looping me into this conversation!  Do we plan to cover during an upcoming Themis meeting?  It looks like the next one is March 20th.  Please let me know if so, and I will be sure to attend.  Thank you!

**Comment by MelaniePhilofsky** [comment]
No, this topic isn't ready for discussion. @crabcakeworld is organizing all interested parties who have use cases and source data. You all will meet to flesh out use cases - what questions are you asking?, what source data do you have (are the data coded with OHDSI supported vocabularies, unsupported vocabularies or free text?), let's see some de-id real data, and explain why the current conventions don't fit your use case. 

When discussing your use cases, decide which ones can we meet with changes to conventions and which ones need a change to the model. We can create conventions now in this WG, but model changes will be in the future when the CDM is updated. @clairblacketer CDM WG will host discussions related to changes to the CDM. Do not worry about domains for concepts, these can be changed.

@AEW0300 has use cases and ideas. Please ensure they are involved. Also, a forum post to ask interested parties to join the effort should be made.

**Comment by AbeckCO** [comment]
Thank you for your response, @MelaniePhilofsky - this is very helpful!  @crabcakeworld please feel free to reach out and we can move forward with these efforts.  Our group (myself, Lisa Schilling, and Heike Düsseldorf) are still interested in this work and have two ongoing use cases to share. 

Thank you again!
